src/Drips.sol

```diff
@@ -13,35 +13,39 @@ struct DripsReceiver {
 /// It's constructed from `amtPerSec`, `start` and `duration` as
 /// `amtPerSec << 64 | start << 32 | duration`.
 /// `amtPerSec` is the amount per second being dripped. Must never be zero.
+/// It's encoded as a fixed point number with a scaling factor of `2^64`.
+/// In other words the amount is multiplied by `2^64` and can have fractions.
 /// `start` is the timestamp when dripping should start.
 /// If zero, use the timestamp when drips are configured.
 /// `duration` is the duration of dripping.
 /// If zero, drip until balance runs out.
-type DripsConfig is uint192;
+type DripsConfig is uint256;
 
 using DripsConfigImpl for DripsConfig global;
 
 library DripsConfigImpl {
     /// @notice Create a new DripsConfig.
     /// @param _amtPerSec The amount per second being dripped. Must never be zero.
+    /// It must be encoded as a fixed point number with a scaling factor of `2^64`.
+    /// In other words the passed value must be multiplied by `2^64` and can have fractions.
```

Comment:
I am trying to understand the pros and cons by defining the scaling factor in `2^64` instead of a power of ten notation.

```
log(10, 2^64) = ~19
```

So obviously, if we define it in power of two we really use all the bits we have available to give the maximum precision to the user with additional 2^64 space.

However, would it not get complicated to use? Instead of defining it with 10^19?

### Full Numbers no Fraction
USDC has 6 decimals. So let say I want to set the `amtPerSec` to 1/10^6= 0.000001 USDC.

**Binary**
In that case my number is 1 and in need to multiple it with the scaling factor

```
1 * 2^64 = 18446744073709551616
```
Would it not be weird to define a full unit as `18446744073709551616`?

**Decimal**
```
1 * 10^19 = 10000000000000000000
```

### Using Fractions
Let say I want to drip  0.0000015 USDC.

I need to calculate
**Binary**
```
1.5 * 2^64 = 27670116110564327424
//let's say they can't use floating operations in javascript and use big integer
1 * 2^64 + 2^64/2 = 27670116110564327424
```
**Decimal**
```
1.5 * 10^19 = 15000000000000000000
```
Assuming I got this right? It would be very hard to check/debug for the developers if the numbers are correct?

Just for a slight bit more precision?

Reply:
I agree that it'd be more dev-friendly to use `10**X`, but this approach is more expensive, to get the full unit you need to divide instead of bitshifting. That's only slightly more expensive, but in the binary search approach it's very important, this operation is repeated tens of thousands of times, so every single gas matters.

Reply:
What would happen if require the user to define it in 10^19 notation but we convert it to a 2^64 representation to get to optimization + user-friendliness?

Just brainstorming here:

```
fullUnits = input/10^19;
fraction = input % 10^19;
input = fullUnits * 2^64 + (fraction/10^19) * 2^64;
```

so 

```
input =  input/10^19 + 2^64 +( (input % 10^19) * 2^64)/10^19;
```

Using the example from above with ` 0.0000015` USDC.

```
// wolfram alpha
15000000000000000000/10^19 + 2^64 +( (15000000000000000000 modulo 10^19) * 2^64)/10^19 = 27670116110564327425.5
```

Compared with the result from above.
27670116110564327424

Reply:
27670116110564327425.5
 27670116110564327424

Reply:
With the lowest possible unit  of `1` instead of `15000000000000000000`.
18446744073709551616
18446744073709551617

Reply:
If it's done on the SDK level, then I'm 100% for it. If you're building a UI for users to enter values, then it's very convenient.

If it's done on the smart contract level, then IMO it's a can of bugs. The users can't pass around exact amounts, to be wei-perfect they need to do additional dancing around this conversion on top of the already complex fractions quirks. We also need to emit events with `10^19` version, because that's what's passed in calldata. That means that the events don't describe the exact dripping configuration, it needs to be converted first. I think that on the protocol level we should stick to a single representation, one way or another.